src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs

```diff
@@ -7607,6 +7607,368 @@ void Method()
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateFieldInExpressionBodiedLocalFunctionInferredFromType()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y()
+    {
+        return Foo();
+
+        int Foo() => [|y|] = 12;
+    }
+}",
+@"class Program
+{
+    private int y;
+
+    public int Y()
+    {
+        return Foo();
+
+        int Foo() => y = 12;
+    }
+}");
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateFieldInBlockBodiedLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return [|y|];
+        }
+    }
+}",
+@"class Program
+{
+    private int y;
+
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return y;
+        }
+    }
+}");
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateReadonlyFieldInBlockBodiedLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return [|y|];
+        }
+    }
+}",
+@"class Program
+{
+    private readonly int y;
+
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return y;
+        }
+    }
+}",
+index: 1);
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGeneratePropertyInBlockBodiedLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return [|y|];
+        }
+    }
+}",
+@"class Program
+{
+    public int y { get; private set; }
+
+    public int Y()
+    {
+        return Foo();
+
+        int Foo()
+        {
+            return y;
+        }
+    }
+}",
+index: 2);
+        }
+
+        [WorkItem(26993, "https://github.com/dotnet/roslyn/issues/26993")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task TestGenerateFieldInBlockBodiedLocalFunctionInferredFromType()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    public int Y()
+    {
+        return Foo();
+
+        int Foo() 
+        {
+            return [|y|] = 12;
+        }
+    }
+}",
+@"class Program
+{
+    private int y;
+
+    public int Y()
```

Comment:
Please use another term than "Foo" in all the added tests. Thanks
`Method()`, `local()` or `property` would be fine.

Reply:
Using `property` is not possible unfortunately, no code actions are available then. Which might actually be a bug, since it's not a reserved keyword as far as I know. Creating the property manually with the identifier `property` is just fine. Do you agree?

I'll just call it `prop` for now (and I'm working on all the other review comments now as well, I hope to finish it later today!).

Reply:
Just call it P or whatever, it doesn't matter at all. It's not that `Foo` is discouraged because it's not descriptive. It's just a bad word for some reason and doesn't pass some kind of MS internal check.

Reply:
I know, however, since we now have the opportunity to make the identifiers more descriptive as well, why not? I already did @jcouv's other rename suggestions too.